components/public-api-server/pkg/apiv1/workspace_test.go

```diff
@@ -247,6 +247,62 @@ func TestWorkspaceService_DeleteWorkspace(t *testing.T) {
 	})
 }
 
+func TestWorkspaceService_UpdateWorkspaceUserPin(t *testing.T) {
+
+	// workspaceID := workspaceTestData[0].Protocol.Workspace.ID
+
+	t.Run("invalid argument when workspace ID is missing", func(t *testing.T) {
+		_, client := setupWorkspacesService(t)
+
+		_, err := client.UpdateWorkspaceUserPin(context.Background(), connect.NewRequest(&v1.UpdateWorkspaceUserPinRequest{
+			WorkspaceId: "",
+			Pinned:      true,
+		}))
+		require.Error(t, err)
+		require.Equal(t, connect.CodeInvalidArgument, connect.CodeOf(err))
+	})
+
+	t.Run("invalid argument when workspace ID does not validate", func(t *testing.T) {
+		_, client := setupWorkspacesService(t)
+
+		_, err := client.UpdateWorkspaceUserPin(context.Background(), connect.NewRequest(&v1.UpdateWorkspaceUserPinRequest{
+			WorkspaceId: "some-random-not-valid-workspace-id",
+			Pinned:      true,
+		}))
+		require.Error(t, err)
+		require.Equal(t, connect.CodeInvalidArgument, connect.CodeOf(err))
+	})
+
+	// t.Run("not found when workspace does not exist", func(t *testing.T) {
+	// 	serverMock, client := setupWorkspacesService(t)
+
+	// 	serverMock.EXPECT().UpdateWorkspaceUserPin(gomock.Any(), workspaceID, protocol.PinActionPin).Return(&jsonrpc2.Error{
+	// 		Code:    404,
+	// 		Message: "not found",
+	// 	})
+
+	// 	_, err := client.UpdateWorkspaceUserPin(context.Background(), connect.NewRequest(&v1.UpdateWorkspaceUserPinRequest{
+	// 		WorkspaceId: workspaceID,
+	// 		Pinned:      true,
+	// 	}))
+	// 	require.Error(t, err)
+	// 	require.Equal(t, connect.CodeNotFound, connect.CodeOf(err))
+	// })
+
+	// t.Run("delegates to server", func(t *testing.T) {
+	// 	serverMock, client := setupWorkspacesService(t)
+
+	// 	serverMock.EXPECT().UpdateWorkspaceUserPin(gomock.Any(), workspaceID, protocol.PinActionPin).Return(nil)
+
+	// 	resp, err := client.UpdateWorkspaceUserPin(context.Background(), connect.NewRequest(&v1.UpdateWorkspaceUserPinRequest{
+	// 		WorkspaceId: workspaceID,
+	// 	}))
+	// 	require.NoError(t, err)
+
+	// 	requireEqualProto(t, &v1.UpdateWorkspaceUserPinResponse{}, resp.Msg)
+	// })
```

Comment:
I am stuck on the tests here. I am getting the following fail:

```
--- FAIL: TestWorkspaceService_UpdateWorkspaceUserPin (0.35s)
    --- FAIL: TestWorkspaceService_UpdateWorkspaceUserPin/delegates_to_server (0.35s)
        /workspace/gitpod/components/public-api-server/pkg/apiv1/workspace.go:309: Unexpected call to *protocol.MockAPIInterface.UpdateWorkspaceUserPin([context.Background.WithValue(type *http.contextKey, val <not Stringer>).WithValue(type *http.contextKey, val 127.0.0.1:33343).WithCancel.WithCancel.WithValue(type auth.contextKey, val <not Stringer>) gitpodio-gitpod-isq6xj458lj 0xc000310280]) at /workspace/gitpod/components/public-api-server/pkg/apiv1/workspace.go:309 because: 
            expected call at /workspace/gitpod/components/public-api-server/pkg/apiv1/workspace_test.go:295 doesn't match the argument at index 2.
            Got: 0xc000310280 (*protocol.PinAction)
            Want: is equal to pin (protocol.PinAction)
        /workspace/gitpod/components/public-api-server/pkg/apiv1/workspace_test.go:300: 
            	Error Trace:	/workspace/gitpod/components/public-api-server/pkg/apiv1/workspace_test.go:300
            	Error:      	Received unexpected error:
            	            	unavailable: EOF
            	Test:       	TestWorkspaceService_UpdateWorkspaceUserPin/delegates_to_server
        /workspace/gitpod/components/public-api-server/pkg/apiv1/controller.go:269: missing call(s) to *protocol.MockAPIInterface.UpdateWorkspaceUserPin(is anything, is equal to gitpodio-gitpod-isq6xj458lj (string), is equal to pin (protocol.PinAction)) /workspace/gitpod/components/public-api-server/pkg/apiv1/workspace_test.go:295
        /workspace/gitpod/components/public-api-server/pkg/apiv1/controller.go:269: aborting test due to missing call(s)
FAIL
FAIL	github.com/gitpod-io/gitpod/public-api-server/pkg/apiv1	0.355s
```

@easyCZ have you seen something like this? Am I passing the arguments incorrectly within `serverMock.EXPECT().UpdateWorkspaceUserPin`?